tests/test_google_ads_interrupted_sync_add_stream.py

```diff
@@ -0,0 +1,269 @@
+                interrupted_record_count = len(interrupted_records)
+
+                if expected_replication_method == self.INCREMENTAL:
+
+                    # gather expectations
+                    expected_primary_key = list(self.expected_primary_keys()[stream])[0]
+                    expected_replication_key = list(self.expected_replication_keys()[stream])[0]  # assumes 1 value
+                    testable_customer_ids = set(self.get_customer_ids()) - {'2728292456'}
+                    for customer in testable_customer_ids:
+                        with self.subTest(customer_id=customer):
+
+                            # gather results
+                            start_date_datetime = dt.strptime(self.start_date, self.START_DATE_FORMAT)
+                            oldest_record_datetime = dt.strptime(interrupted_records[0].get(expected_replication_key), self.REPLICATION_KEY_FORMAT)
+                            final_stream_bookmark = final_state['bookmarks'][stream]
+                            final_bookmark = final_stream_bookmark.get(customer, {}).get(expected_replication_key)
+                            final_bookmark_datetime = dt.strptime(final_bookmark, self.REPLICATION_KEY_FORMAT)
+
+                            # Verify final bookmark saved match formatting standards for resuming sync
+                            self.assertIsNotNone(final_bookmark)
+                            self.assertIsInstance(final_bookmark, str)
+                            self.assertIsDateFormat(final_bookmark, self.REPLICATION_KEY_FORMAT)
+
+                            if stream in full_sync_state['bookmarks'].keys():
+
+                                full_sync_stream_bookmark = full_sync_state['bookmarks'][stream]
+                                full_sync_bookmark = full_sync_stream_bookmark.get(customer, {}).get(expected_replication_key)
+                                full_sync_bookmark_datetime = dt.strptime(full_sync_bookmark, self.REPLICATION_KEY_FORMAT)
+                                final_stream_bookmark = final_state['bookmarks'][stream]
+                                final_bookmark = final_stream_bookmark.get(customer, {}).get(expected_replication_key)
+                                final_bookmark_datetime = dt.strptime(final_bookmark, self.REPLICATION_KEY_FORMAT)
+
+                            if stream in interrupted_state['bookmarks'].keys():
+
+                                interrupted_stream_bookmark = interrupted_state['bookmarks'][stream]
+                                interrupted_bookmark = interrupted_stream_bookmark.get(customer, {}).get(expected_replication_key)
+                                interrupted_bookmark_datetime = dt.strptime(interrupted_bookmark, self.REPLICATION_KEY_FORMAT)
+
+                                # Verify state ends with the same value for common streams after both full and interrupted syncs
+                                if stream != added_stream:
```

Comment:
The `if` is not needed since `added_streams` is not in interrupted_bookmarks

Reply:
Refactored